File: backend/LogisticsAssetTracker.Api/Domain/Entities/AuditLog.cs
namespace LogisticsAssetTracker.Api.Domain.Entities;

public class AuditLog
{
    public Guid Id { get; set; }
    public Guid? UserId { get; set; }
    public User? User { get; set; }
    public string Action { get; set; } = string.Empty;
    public string EntityType { get; set; } = string.Empty;
    public Guid? EntityId { get; set; }
    public string? MetadataJson { get; set; }
    public DateTime CreatedAt { get; set; }
    public bool IsSuspicious { get; set; }
    public string? SuspiciousReason { get; set; }
}

File: frontend/src/types/auditLog.ts
// Document 04/06/07: audit entries never carry secrets. metadataJson is a raw JSON
// string captured by the backend for traceability (previous/attempted values, reasons).
export interface AuditLog {
  id: string;
  userId?: string | null;
  userName?: string | null;
  action: string;
  entityType: string;
  entityId?: string | null;
  metadataJson?: string | null;
  createdAt: string;
  isSuspicious: boolean;
  suspiciousReason?: string | null;
}

export interface AuditLogListQuery {
  userId?: string;
  entityType?: string;
  entityId?: string;
  action?: string;
  isSuspicious?: boolean;
  fromDate?: string;
  toDate?: string;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortDirection?: "asc" | "desc";
}
